packages/backend/server/src/modules/workspaces/controller.ts

```diff
@@ -104,4 +105,51 @@ export class WorkspacesController {
     res.send(update);
     this.logger.debug(`workspaces doc api: ${format(process.hrtime(start))}`);
   }
+
+  @Get('/:id/docs/:guid/histories/:timestamp')
+  @Auth()
+  async history(
+    @CurrentUser() user: UserType,
+    @Param('id') ws: string,
+    @Param('guid') guid: string,
+    @Param('timestamp') timestamp: string,
+    @Res() res: Response
+  ) {
+    const docId = new DocID(guid, ws);
+    let ts;
+    try {
+      const timeNum = parseInt(timestamp);
+      if (Number.isNaN(timeNum)) {
+        throw new Error('Invalid timestamp');
+      }
+
+      ts = new Date(timeNum);
+    } catch (e) {
+      throw new Error('Invalid timestamp');
+    }
+
+    if (
+      // if a user has the permission
+      !(await this.permission.isAccessible(
```

Comment:
isAccessible only check if has Read permission, we need > Admin

Reply:
we're doing the permission controlling badly. actually all users with `Read` permission can edit the docs, if we limit the history functionality to `Write` or `Admin`, users may complain about it